### ISSUE TITLE
[RFC]: Audit and clean up bundled deploy YAML arguments and documentation

### ISSUE THREAD

**Issue by hsliuustc0106** [vllm-project/vllm-omni#6176]
### Motivation

This is a focused follow-up to #2072 and #1807.

A static audit of the current bundled deploy configurations found that many YAML arguments were mechanically preserved when config-schema defaults changed to `None` (notably in `b076006c`). Some are now genuinely dead or overwritten, some only duplicate current defaults, and others still preserve important model or hardware behavior.

We should clean up the files without treating every repeated value as redundant.

Audit baseline: `origin/main` at `4b66f62e10f1b765dc6dc6baa5bbfe633234ec25`.

Scope: 19 profiles across:

- Qwen3-Omni
- Qwen3-TTS, including the high-concurrency profile and separate forced-aligner config
- HunyuanImage 3.0
- GLM-Image
- MiniCPM-o 4.5
- VoxCPM2

This was a source/config/history audit only. No GPU/NPU/ROCm/XPU/MUSA workload was run, so capacity, graph, cache, and hardware-layout changes remain validation-gated.

### Summary of findings

| Family | Pipeline | Source-proven cleanup candidates | Validate or reconcile first |
|---|---|---|---|
| Qwen3-Omni | Thinker → Talker → Code2Wav | Greedy `top_p`/`top_k`; Mori greedy seed; Mori RDMA-only fields under XGMI | Trust/prefix/capacity pins; Mori sampling, initial-chunk, and long-text drift |
| Qwen3-TTS | Talker → Code2Wav | Dead `codec_streaming` and connector polling keys; Code2Wav sampling block | Talker sampling, initial chunk, left context, graph/concurrency settings |
| HunyuanImage | AR → DiT, plus standalone variants | Legacy YAML topology metadata; unused connectors; greedy filters; invalid diffusion-stage engine args | KV reuse, MRoPE, prefetch, parallel and platform profiles |
| GLM-Image | AR → DiT/VAE | Diffusion `max_num_batched_tokens`/prefix key; overwritten NPU scheduler; identity parallel values | AR sampling/EOS/length; NPU topology; trust policy |
| MiniCPM-o | Thinker → Talker → Code2Wav | Dead connector polling keys; empty platform blocks; pipeline-overwritten sampling fields | Multimedia parity, graph/codec settings, duplex and replica layouts |
| VoxCPM2 | Single AR stage with internal audio path | Forced-false async flag; KV-ignored GPU utilization; unreachable optimization; greedy no-ops | Explicit KV/concurrency/graph/precision pins; trust policy |

#### Qwen3-TTS

In both `qwen3_tts.yaml` and `qwen3_tts_high_concurrency.yaml`, remove:

```yaml
codec_streaming: true
connector_get_sleep_s: 0.01
connector_get_max_wait_first_chunk: 3000
connector_get_max_wait: 300
```

`SharedMemoryConnector` does not read these fields. Streaming is selected by top-level `async_chunk: true`; the transfer adapter uses its own waiting behavior.

Also remove the Code2Wav `default_sampling_params` blocks. The generation worker does not run token logits/sampling, so these parameters do not control Code2Wav execution.

Keep the Talker sampling contract, `min_tokens: 2`, codec left context, initial chunk behavior, batch limit, graph buckets, and capacity settings.

The same Code2Wav sampling finding should be checked in embedded Qwen3-TTS stages such as Aura.

#### Qwen3-Omni

- Remove Code2Wav `top_p: 1.0` and `top_k: -1` under `temperature: 0`.
- In the Mori profile, also remove the greedy seed.
- Remove Mori RDMA-only `device_name`, `qp_per_transfer`, `post_batch_size`, and `num_worker_threads` when using the XGMI backend.
- Optionally omit Mori values equal to current connector defaults after deciding the “pin defaults vs only deviations” policy.

Mori must otherwise be reconciled before cleanup: it claims parity with the main profile but retains old stochastic Thinker sampling, omits `initial_codec_chunk_frames: 4`, and missed the Stage-2 65536-token/chunked-prefill correctness update.

#### HunyuanImage 3.0

- Remove YAML-owned `is_comprehension`, `final_output`, and `final_output_type`; frozen pipeline topology owns these values.
- Remove unreferenced Yuanrong and SharedMemory connector definitions from the combined profile.
- Remove greedy AR `top_p`/`top_k`.
- Review the duplicated `edges` block: connector mappings already define the active edge, and current connector initialization does not consume its window/inflight values.
- Remove fields filtered from diffusion configuration, including diffusion-stage `max_num_batched_tokens`, plus exact identity/default parallel settings where policy permits.

There is a separate correctness question: the YAML says `is_comprehension: true` is required, but the registry currently owns `owns_tokenizer=False`. If tokenizer ownership is required, fix the registry rather than retaining an ineffective YAML key.

#### GLM-Image

- Remove Stage-1 `max_num_batched_tokens` and `enable_prefix_caching`; they are not owned by `OmniDiffusionConfig`.
- Remove NPU Stage-0 `sequence_parallel_size: 1`.
- Remove the explicit NPU `scheduler_cls: OmniARScheduler`; scheduler resolution from `async_scheduling: true` overwrites it with the async scheduler.
- On NPU DiT, remove identity `tensor_parallel_size: 1`, `ring_degree: 1`, and derived `sequence_parallel_size: 4`; retain `ulysses_degree: 4`.

Keep `async_chunk: false`, AR sampling, stop token, maximum generation length, device/HCCL layout, TP4, NPU worker, and compilation settings.

#### MiniCPM-o 4.5

Across the family:

- Remove the three unread `connector_get_*` polling keys.
- Remove empty `platforms: {}` blocks.
- Remove `detokenize` values overwritten by frozen pipeline sampling constraints.
- Remove greedy filters/seeds and generation-stage sampling values only after adding resolved-config equivalence coverage.

Before normalizing variants, resolve:

- Default Stage-0 repetition penalty `1.0` versus `1.1` in sibling profiles.
- Inconsistent multimodal/interleave limits.
- The “8x4090” base uses six GPUs, which should be documented as a host profile if intentional.
- The 8x Stage-1-replica overlay places Talker replicas on GPUs 4–7 while inherited Code2Wav remains on GPU 5, creating likely memory oversubscription.

#### VoxCPM2

- Remove `async_chunk: false`; single-stage merge forces it off.
- Remove `gpu_memory_utilization: 0.9`; explicit `kv_cache_memory_bytes` bypasses utilization-based KV sizing.
- Remove `enable_loc_dit_reduce_overhead_no_cg: true`; the unified decode graph branch makes it unreachable.
- Under `temperature: 0`, remove `top_p`, `top_k`, `seed`, and repetition penalty `1.0`.

Most `voxcpm2_runtime_config` entries duplicate current model defaults. Whether to retain them depends on the repository policy below. Keep the non-default unified graph batch cap of 8.

### Configuration policy to document

Before broad cleanup, define whether bundled YAMLs:

1. contain only deviations from runtime/model defaults, or
2. deliberately freeze all validated behavior even when it equals today's defaults.

Recommended policy:

- **Topology-owned:** pipeline registry only; reject or warn on topology keys in deploy YAML.
- **Model behavior:** keep explicit when it defines quality, stopping, streaming, or output semantics.
- **Hardware profile:** keep explicit and label validation hardware.
- **Backend-derived defaults:** omit unless deliberately pinned; document that omission delegates to the installed backend.
- **Security/loading:** treat `trust_remote_code` changes as startup-matrix work, not cosmetic cleanup.

Uniform pipeline-wide settings such as `trust_remote_code` and `enable_prefix_caching` should live at top level. Hoisting is behavior-preserving; deleting is not necessarily behavior-preserving.

### Documentation work

Update `docs/configuration/stage_configs.md` before or with the cleanup:

- Current docs present `trust_remote_code=true`, prefix caching false, and several capacity values as schema defaults. Current code uses `None` and delegates downstream.
- Explain merge/override precedence, including `base_config` and platform overlays.
- Explain that pipeline sampling constraints override deploy YAML sampling defaults.
- Explain that generation/Code2Wav stages may not perform token sampling.
- Document model/adapter-owned connector extras instead of claiming SharedMemory has no extras.
- Clarify that unknown keys are not universally forwarded; typed and diffusion configs may filter unowned fields.
- Separate forced-aligner configuration from normal `DeployConfig`.
- Clearly isolate the legacy `stage_args` reference.

Also fix model-specific drift found during the audit:

- Qwen3-Omni and Qwen3-TTS comments contradict current token budgets/prefix settings.
- Qwen3-Omni Mori documentation claims false parity and contains stale frame-rate/RDMA guidance.
- GLM recipe uses inconsistent ports and output filenames and conflicts on seed/hardware claims.
- MiniCPM docs describe an obsolete two-stage/default-two-GPU setup and stale memory/video values.
- VoxCPM2 recipe documents old concurrency/KV behavior and recommends changing GPU utilization even though explicit KV bytes take precedence.

### Tests required

Before cleanup:

- Add CPU-only resolved-config snapshots or semantic comparisons for all scoped profiles.
- Compare effective stage engine args, sampling constraints, connector configuration, and platform overlays before/after removal.
- Assert that unsupported/unowned fields fail or warn instead of silently surviving parsing.
- Add Mori and GLM NPU resolution tests.
- Add coverage for `base_config` variant inheritance and top-level-vs-stage precedence.
- Ensure tests validate connector consumption rather than merely asserting that arbitrary extras survive parsing.

Hardware/quality validation is required before removing or changing:

- `trust_remote_code`
- `enable_prefix_caching`
- capacity and memory limits
- eager/graph/compilation settings
- TP/SP/EP/device profiles
- model sampling values that are not proven greedy no-ops

### Proposed implementation order

- [ ] Correct the deploy-configuration guide and define the omission/pinning policy.
- [ ] Add resolved-config contract tests for the scoped YAMLs.
- [ ] Remove source-proven dead, filtered, overwritten, and unreachable fields.
- [ ] Hoist uniform pipeline-wide settings.
- [ ] Convert suitable full-copy variants to thin `base_config` overlays.
- [ ] Reconcile Mori, MiniCPM multimedia parity, and replica-device conflicts.
- [ ] Run model/platform startup, quality, memory, and performance validation for the validate-first set.
- [ ] Expand the audit to the remaining bundled deploy files once the policy and tests are established.

### Acceptance criteria

- Bundled deploy YAMLs contain no keys without an identified runtime/config consumer.
- Frozen topology is represented only by pipeline registry definitions.
- Exact-default pins are either removed or annotated as intentional compatibility/validation contracts.
- All variants have resolved-config parity tests.
- Documentation matches actual schema defaults and precedence.
- Hardware-specific settings are changed only with recorded validation evidence.
- No user-visible model quality, streaming, stopping, memory, or platform behavior changes unintentionally.

### Any Other Things

Related issues: #2072, #1807, and #1462.

The full static audit deliberately distinguishes source-proven redundancy from current-default equivalence and hardware-dependent behavior.

**Comment by hsliuustc0106** [comment]
@linyueqian @alex-jw-brooks @NickCao @Bounty-hunter @psv666 PTAL
